### DISCUSSION TITLE
Add non-generic interfaces IDataPortal and IChildDataPortal

### DISCUSSION THREAD

**Discussion by hoangcuongvn** [MarimerLLC/csla#2916] [Ideas]
@rockfordlhotka Can you add non-generic interfaces `IDataPortal` and `IChildDataPortal`? They would look something like the code underneath. Because sometimes we have to deal with reflection, those non-generic interfaces would help us much. In other hand, they are similar to the former static class  DataPortal in Csla 5. That would help migrating code from Csla 5 to Csla 6 much easier.
```
public interface IDataPortal
{
  Task<object> CreateAsync(Type type, params object[] criteria);
  Task<object> FetchAsync(Type type, params object[] criteria);
  Task<object> UpdateAsync(object obj);
  ...
}
```

**Comment by rockfordlhotka** [comment]
It seems like this would be in place of the change suggested in #2915?

**Reply by hoangcuongvn** [reply]
Yes, the non-generic interfaces would resolve the problem in #2915 👍

**Reply by rockfordlhotka** [reply]
Done in #2918
